cdbot/cogs/admin.py

```diff
@@ -89,6 +92,32 @@ async def on_member_join(self, member: Member):
             # assign placeholder nickname
             await member.edit(nick=PLACEHOLDER_NICKNAME)
 
+    @Cog.listener()
+    async def on_message(self, message):
+        def check(m):
+            return m.author == message.author and (datetime.datetime.utcnow() - m.created_at).seconds < 4
+
+        def who(m):
+            return m.author == message.author
+
+        if not message.author.bot:
+            # Checks if message contains banned word.
+            if any([word in message.content for word in BANNED_WORDS]):
```

Comment:
This will block `snigger`. Consider using a regex of `\Wword\W`.

Reply:
Surrounding the generator in `[]` is redundant, and is marginally slower than just leaving it as a generator; it defeats `any`'s early-exit handler. Likewise in other places `any([...])` has been used.